### PR TITLE
Add support for NULS addresses

### LINKED ISSUE #199: Add support for NULS (NULS) addresses in encoding library
Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

We will likely be reviewing PRs and paying out bounties more or less on a weekly basis. No need to ping us to check your PR, we will definitely get to it.

Thanks for your help!

### PR DESCRIPTION
**Pull request by Echolon166**
## Issue number

Closes #199 

## Description

Add support for NULS addresses

## Reference to the specification

- https://github.com/nuls-io/nuls-v2-js-sdk/blob/cc208cdc153d3fca1e001fbe7c457780523bd96a/lib/api/sdk.js#L85
- https://github.com/nuls-io/nuls-v2-js-sdk/blob/cc208cdc153d3fca1e001fbe7c457780523bd96a/lib/api/sdk.js#L107
- https://github.com/nuls-io/nuls-v2-js-sdk/blob/cc208cdc153d3fca1e001fbe7c457780523bd96a/lib/api/sdk.js#L138

## Reference to the test address.

- Test cases are 1st and 2nd from [Nuls Explorer](https://nulscan.io/address).

## List of features added/changed

- Implement NULS encoder
- Implement NULS decoder

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.
- [x] I have specified correct coinTypes specified at [Slip 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- [x] I have provided the reference link to the specification I implemented.
- [x] I have provided enough explanation about how I provided the test address

### CODE REVIEW

**Comment by Echolon166** [comment]
In [tslint](https://palantir.github.io/tslint/rules/no-bitwise/), bitwise operations are not allowed because they are often typos.
Since they are used intentionally in this case, I've disabled no-bitwise tslint rule for the nuls encoder&decoder.

File: src/index.ts [hunk 1]
```diff
@@ -993,6 +993,66 @@ function aionEncoder(data: Buffer): string {
...
+    temp = temp > 127 ? temp - 256 : temp;
+    tempBuffer[i] = temp;
+    xor ^= temp;
+  }
+  tempBuffer[data.length] = xor;
+
+  let prefix = "";
+  if(1 === chainId) {
+    prefix = 'NULS';
+  }
```
Comment (makoto) [inline src/index.ts:1014]: Can you also support non production prefixes? 

https://github.com/nuls-io/nuls-v2-js-sdk/blob/cc208cdc153d3fca1e001fbe7c457780523bd96a/lib/api/sdk.js#L122

Comment (Echolon166) [reply src/index.ts]: Done.

Comment (makoto) [reply src/index.ts]: Can you also add non-production prefixes to the test?

**Comment by makoto** [comment]
Just a reminder that this still needs test case for test prefixes

File: src/index.ts (hunk 1 above)
Comment (Echolon166) [reply src/index.ts]: Done. Test cases are from: https://github.com/nuls-io/nuls-v2/blob/0d55b325438307319cc2bd385f4064c4518f705b/module/nuls-account/src/test/java/io/nuls/account/util/AddressToolTest.java

**Pull request merged**